### PR TITLE
Add a function to get a look by a look_id

### PR DESCRIPTION
**Pull request by peterhurford**

### CODE REVIEW

**Comment by robertzk** [comment]
@peterhurford Let me know when this is ready to go and @abelcastilloavant and I will take a look.

File: R/run_look.R [hunk 1]
```diff
@@ -0,0 +1,35 @@
+#' Pull a look from a look id.
+#'
+#' @param look_id integer.
+#' @param format character. The format to use. Defaults to json.
+#' @param limit numeric. The \code{limit} parameter of the query.
+#' @return a data.frame containing the data returned by the query
+#' @export
+run_look <- function(look_id, format = "json", limit = 1000) {
+  # TODO: DRY
+  env_var_descriptions <- list(
```
Comment (abelcastilloavant) [inline R/run_look.R]: the helper in https://github.com/avantcredit/looker3/blob/master/R/utils.R#L15 should help you DRY this a bit. :)

**Comment by robertzk** [comment]
Are we doing anything with this PR? @peterhurford @abelcastilloavant

**Comment by peterhurford** [comment]
@robertzk I no longer have access to a Looker instance so it is hard for me to test it live. I think all I need to do though is add a unit test and DRY up the code, which shouldn't be too hard. It's still on my radar, but lower priority.

**Comment by robertzk** [comment]
Thanks @peterhurford -- I've pushed up my suggested changes to the [looker_project2 branch](https://github.com/avantcredit/looker3/compare/looker_project2?expand=1). Feel free to merge those in.

**Comment by peterhurford** [comment]
Thanks @robertzk -- looks like https://github.com/avantcredit/looker3/commit/58ae9e80927fe4f417f9a1b243a19565f245dcbc takes care of DRY. Maybe we could cherry pick that commit (plus your redoc) into this branch?

Should we also add a unit test?

**Comment by robertzk** [comment]
@peterhurford I don't have write access to your branch but go for it. We should add a test if you have time but I'm comfortable merging whenever.

**Comment by peterhurford** [comment]
@robertzk I have access to a looker instance again!

**Comment by robertzk** [comment]
@peterhurford Awesome! Let me know if your manual testing works and I'll merge when we have a unit test.

**Comment by peterhurford** [comment]
@robertzk @abelcastilloavant I DRY'd and added unit tests. Good to merge?

**Comment by iveksl2** [comment]
👍 #collaboration #mutually-beneficial

**Comment by abelcastilloavant** [comment]
CI is reporting test failures for `run_look` - probably easy to fix by mocking `setup_using_env_vars`

Good to go once that's fixed. 👍 

```
1. Error: run_look can get a look (@test-run_look.R#4) -------------------------
Your environment variables are not set correctly. please place your Looker 3.0 API url in the environment variable LOOKER_URL.
1: with_mock(`looker3:::login_api_call` = function(...) NULL, `looker3:::logout_api_call` = function(...) NULL, 
       `looker3:::cached_token_is_invalid` = function(...) FALSE, `looker3:::pull_look_api_call` = function(...) NULL, 
       `httr::content` = function(...) iris, {
           expect_identical(run_look(1), iris)
       }) at testthat/test-run_look.R:4
2: eval(expression, parent.frame())
3: eval(expr, envir, enclos)
4: expect_identical(run_look(1), iris)
5: identical(object, expected)
6: run_look(1)
7: setup_using_env_vars()
8: lapply(names(env_var_descriptions), function(name) {
       env_var <- Sys.getenv(name)
       if (env_var == "") {
           stop(paste0("Your environment variables are not set correctly. ", "please place your Looker 3.0 ", 
               env_var_descriptions[[name]], " in the environment variable ", name, 
               "."))
       }
       env_var
   })
9: FUN(X[[i]], ...)
10: stop(paste0("Your environment variables are not set correctly. ", "please place your Looker 3.0 ", 
       env_var_descriptions[[name]], " in the environment variable ", name, "."))
2. Error: run_look can trigger a re-login if the cache is invalid (@test-run_look.R#15) 
Your environment variables are not set correctly. please place your Looker 3.0 API url in the environment variable LOOKER_URL.
1: with_mock(`looker3:::login_api_call` = function(...) NULL, `looker3:::logout_api_call` = function(...) NULL, 
       `looker3:::validate_response` = function(...) NULL, `looker3:::cached_token_is_invalid` = function(...) TRUE, 
       `looker3:::pull_look_api_call` = function(...) NULL, `httr::content` = function(...) list(access_token = "TOKENYO", 
           expires_in = 0), {
           token_cache$set("token", list(token = "FAKE_TOKEN"))
           expect_equal(token_cache$get("token")$token, "FAKE_TOKEN")
           run_look(1)
           expect_equal(token_cache$get("token")$token, "TOKENYO")
           token_cache$set("token", list(token = "FAKE_TOKEN"))
           expect_equal(token_cache$get("token")$token, "FAKE_TOKEN")
       }) at testthat/test-run_look.R:15
2: eval(expression, parent.frame())
3: eval(expr, envir, enclos)
4: run_look(1)
5: setup_using_env_vars()
6: lapply(names(env_var_descriptions), function(name) {
       env_var <- Sys.getenv(name)
       if (env_var == "") {
           stop(paste0("Your environment variables are not set correctly. ", "please place your Looker 3.0 ", 
               env_var_descriptions[[name]], " in the environment variable ", name, 
               "."))
       }
       env_var
   })
7: FUN(X[[i]], ...)
8: stop(paste0("Your environment variables are not set correctly. ", "please place your Looker 3.0 ", 
       env_var_descriptions[[name]], " in the environment variable ", name, "."))

```

File: tests/testthat/test-run_look.R [hunk 2]
```diff
@@ -0,0 +1,35 @@
+context("run_look")
+
+test_that("run_look can get a look", {
+  with_mock(
+    `looker3:::login_api_call` = function(...) NULL,
```
Comment (abelcastilloavant) [inline tests/testthat/test-run_look.R]: you want to mock `setup_using_env_vars` here too

Comment (peterhurford) [reply tests/testthat/test-run_look.R]: Done.

**Comment by peterhurford** [comment]
@abelcastilloavant @robertzk Can you advise me on the latest Travis fail? https://travis-ci.org/avantcredit/looker3/jobs/288722751

**Comment by robertzk** [comment]
@peterhurford The R devel build is by self-admission unstable, so I wouldn't worry about it. Just [remove it from `.travis.yml` for us](https://github.com/avantcredit/looker3/blob/master/.travis.yml#L12). Thanks!

**Pull request merged**